### DISCUSSION TITLE
Are Rfc3987 or text-unidecode dependencies?

### DISCUSSION THREAD

**Discussion by Adeline-Toader** [CycloneDX/cyclonedx-python-lib#658] [Q&A] [answered]
Hello,

Does the library rely on Rfc3987 or text-unidecode, even as transitive dependencies? Is there anywhere I can find a list of the dependencies of the library?

Thank you very much,

**Comment by schlenk** [comment]
If this is about licensing, did you already see the discussion about RFC3987 in:

https://github.com/CycloneDX/cyclonedx-python-lib/issues/568 ?

**Comment by jkowalleck** [comment, marked as answer]
the exact (transitive) dependencies of this library are not clear until it is resolved and installed.
this is due to the fact that it has conditional direct dependencies, direct dependencies with version-reigns and version-boundaries, and optional dependencies.

**Reply by jkowalleck** [reply]
you may find the dependencies of this lib the same way you would find the (transitive) dependencies of any other python package - and it completely depends on the installation method and your environment/setup.

**Reply by pombredanne** [reply]
I do not think this is an appropriate answer. Much the same way a vulnerability in your pinned dependency tree is a problem, here I cannot use the format-nongpl extra of the jsonschema by using the jsonschema extra, and this is a problem too, especially since upstream jsonschema offers this option.

**Reply by jkowalleck** [reply]
> Much the same way a vulnerability in your pinned dependency tree is a problem [...]

Comparing apples and bananas.

The dependency in question is an optional one, one that is not installed per default.

> [...] especially since upstream jsonschema offers this option.

It is not that I am ignorant, in fact, I've researched this option multiple times in the past. 
I came to the conclusion, that the non-GPL version are not providing the needed functionality.

Anyway, instead of pointing out one and the same "issue" again, better work with me finding a solution. See, I understand that this might be an issue for some, but for most, it is no issue at all.
I would be happy to see a pullrequest with the needed changes, and a proper (unit-)test for it :-D 

PS: the issue for this PR would be see https://github.com/CycloneDX/cyclonedx-python-lib/issues/743

**Comment by pombredanne** [comment]
This indirect GPL dependency eventually means that it may not be possible to use this library with this format feature in a redistributed deployment that is not using a GPL-compatible license. 

Since jsonschema has a "format-nongpl" options, it would be best to also expose this too, rather than picking as a default the "format" extra that pulls in GPL-licensed code without any warning and on which users have no control.

See https://github.com/python-jsonschema/jsonschema/blob/9c214b1ae56a71a2e0f2523ddb8f2a3335f6186c/pyproject.toml#L61C1-L61C14

The non GPL extra uses https://github.com/python-hyper/rfc3986 instead of which is under an Apache license.

**Reply by jkowalleck** [reply]
> Since jsonschema has a "format-nongpl" options, it would be best to also expose this too,

not feasible. the needed functionality does not exist in the non-GPL version.
